example/ideScripts/utilities.py

```diff
@@ -341,48 +361,107 @@ def getGccIncludePath(gccExePath):
     '''
     Get path to '...\include' folder from 'gccExePath', where standard libs and headers. Needed for VS Code Intellisense.
 
-    If ARM GCC folder structure remains the same as official, .exe is located in \bin folder.
+    If ARM GCC folder structure remains the same as official, the executable is located in \bin folder.
     Other headers can be found in '\lib\gcc\arm-none-eabi\***\include' folder, which is found by searching for
     <stdint.h>.
     '''
     gccExeFolderPath = os.path.dirname(gccExePath)
     gccFolderPath = os.path.dirname(gccExeFolderPath)
     searchPath = os.path.join(gccFolderPath, "lib", "gcc", "arm-none-eabi")
 
-    searchForFile = "stdint.h"
-    for root, dirs, files in os.walk(searchPath, topdown=False):
-        if searchForFile in files:
-            folderPath = pathWithForwardSlashes(root)
-            return folderPath
+    fileName = "stdint.h"
+    folderPath = findFileInFolderTree(searchPath, fileName)
+    if folderPath is None:
+        errorMsg = "Unable to find 'include' subfolder with " + fileName + " file on path:\n\t"
+        errorMsg += searchPath
+        printAndQuit(errorMsg)
+
+    return folderPath
+
+def getPython3Path():
+    '''
+    Uses detectOs() to determine the correct python command to use for python related tasks
+    '''
+    osIs = detectOs()
+
+    if osIs == "unix" or osIs == "wsl": # detected unix based system
+        pythonPath = "python3"
+    else: # windows or other system
+        pythonPath = "python"
+
+    if not pathExists(pythonPath):
+        msg = "\n\tPython version 3 or later installation not detected, please install or enter custom path below."
+        print(msg)
+        pythonPath = getUserPath(pythonPath)
 
-    errorMsg = "Unable to find 'include' subfolder with " + searchForFile + " file on path:\n\t"
-    errorMsg += searchPath
-    printAndQuit(errorMsg)
+    return pythonPath
 
 
-def getSTLinkPath(openOCDTargetPath):
+def getOpenOcdConfig(openOcdPath):
     '''
-    Get path to '.../scripts/interface/stlink.cfg' file from 'openOCDTargetPath'
+    Get openOCD configuration from user, eg. '-f interface/stlink.cfg -f target/stm32f0x.cfg'

```

Comment:
I suppose it depends on the point of view. As someone who has worked with OpenOCD a fair amount I have a few of the scripts memorised, so it is easier to just write the command. There are issues with the old method too, since the number of configuration files isn't always two. I agree for most users it would be easier to just specify two configuration files - i'll give this a look.

Reply:
Oh, I don't have that much experience with OpenOCD - never had a case with more than two configuration files, so I didn't even know that that could be the case. Anyway, currently implemented concept of getting multiple files seems a good solution.